tsdb/chunks/chunks.go

```diff
@@ -252,34 +257,58 @@ func MergeOverlappingChunks(chks []Meta) ([]Meta, error) {
 			last++
 			continue
 		}
-		nc := &newChks[last]
-		if c.MaxTime > nc.MaxTime {
-			nc.MaxTime = c.MaxTime
-		}
-		chk, err := MergeChunks(nc.Chunk, c.Chunk)
+		dividedChks, err := MergeChunks(newChks[last].Chunk, c.Chunk, MaxSamplesInChunk)
 		if err != nil {
 			return nil, err
 		}
-		nc.Chunk = chk
+		newChks[last] = dividedChks[0]
```

Comment:
All of this can be replaced with single line (: 

```
newChks = append(newChks[:len(newChks)-1], dividedChks...)
```

Reply:
Yes, you are right. Thank you!